File: SentiLab-kotlin-android/SentiHandler/app/src/main/java/com/nextop/project/sentihandler/MainActivity.java
package com.nextop.project.sentihandler;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private int _cnt = 0;
    private TextView tv;
    private Handler _handler = new Handler();
    private Runnable _runnable = new Runnable() {
        @Override
        public void run() {
            _cnt++;
            tv.setText("" + _cnt);

            if(_cnt < 5) {
                _handler.postDelayed(this, 1000); // 1000밀리 세컨드이후 post
            } else{
                _handler.removeCallbacks(_runnable);
            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv = findViewById(R.id.act_main_tv);

        _handler.postDelayed(_runnable,1000); // 1000밀리 세컨드이후 post
    }
}

File: SentiLab-kotlin-android/FutureTask/app/src/main/java/com/nextop/project/futuretask/MainActivity.kt
package com.nextop.project.futuretask

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import java.util.concurrent.Executor
import java.util.concurrent.Executors
import java.util.concurrent.FutureTask

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun futureTask(){
        // 1. FutureTask => Thread
        val ftCal = FutureTask{ // 타입추론 적용됨
            return@FutureTask "Result Runnable"
        }
        val ftRun = FutureTask({

        },"Result Callable")

        // 2. ExecutorService
        val exs = Executors.newFixedThreadPool(2)
        exs.execute(ftCal)
        exs.execute(ftRun)

        Log.e("MainActivity","" + ftCal.get())
        Log.e("MainActivity","" + ftRun.get())
    }
}
